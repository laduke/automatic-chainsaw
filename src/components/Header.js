import { connect } from 'react-redux';
import React from 'react';
import {
  setVisibilityFilter,
  SHOW_ALL,
  SHOW_FAVORITE
} from '../actions';

const highlight = (a, b) => {
  return a === b ? {backgroundColor: 'teal'} : {};
};

export const Description = ({description}) => <p>{description}</p>;
export const PublicationDate = ({pubDate}) => <p>{pubDate}</p>;
export const Title = ({title}) => <h3>{title}</h3>; 
export const FilterLinks = ({visibilityFilter, onFilterClick}) => (
  <p>
    <span
      style={highlight(visibilityFilter, SHOW_ALL)}
      onClick={() => onFilterClick(SHOW_ALL)}>show all </span>
    <span
      style={highlight(visibilityFilter, SHOW_FAVORITE)}
      onClick={() => onFilterClick(SHOW_FAVORITE)}>show favorites </span>
  </p>
);

const Header = (state) => {
  const {description, pubDate, title} = state.header;
  const {onFilterClick, loading} = state;
  const {visibilityFilter} = state.userData;

  return (
    <div>
      <Title title={title} />
      <Description description={description} />
      <PublicationDate pubDate={pubDate} />
      <FilterLinks
        visibilityFilter={visibilityFilter}
        onFilterClick={onFilterClick} />
      <Loading loading={loading}/>
    </div>
    );
};

export const Loading = ({loading}) => {
  if(loading) {
    return <h4 style={{position: 'fixed', top: 0, right: 10}}> loading </h4>;
  } else {
    return null; 
  }
};

const mapStateToProps = ({ndbc}) => {
  const {loading, header, userData} = ndbc;
  return {
    loading,
    header,
    userData
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
      onFilterClick: (filter) => {
        dispatch(setVisibilityFilter(filter));
      }
  };
};

const ConnectedHeader = connect(mapStateToProps, mapDispatchToProps)(Header);
export default ConnectedHeader;
