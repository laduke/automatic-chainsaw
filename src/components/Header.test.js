import React from 'react';
import { Loading, Description, PublicationDate, Title, FilterLinks } from './Header';
import renderer from 'react-test-renderer';

test('Loading appears when buoy data is loading', () => {
  const component = renderer.create(
    <Loading loading={true}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();


  const nullComponent = renderer.create(
    <Loading loading={false}/>
  );
  tree = nullComponent.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Description renders', () => {
  const component = renderer.create(
    <Description description={'Hello, this is the description'}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Publication Date renders', () => {
  const component = renderer.create(
    <PublicationDate pubDate={'Fri, 09 Dec 2016 00:16:24 +0000'}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Title renders', () => {
  const component = renderer.create(
    <Title title={'NDBC - Station Observations near 40N 73W'}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('FilterLinks render with feedback', () => {
  const componentAll = renderer.create(
    <FilterLinks visibilityFilter={'SHOW_ALL'}/>
  );
  let tree = componentAll.toJSON();
  expect(tree).toMatchSnapshot();

  const componentFavorite = renderer.create(
    <FilterLinks visibilityFilter={'SHOW_FAVORITE'}/>
  );
  tree = componentFavorite.toJSON();
  expect(tree).toMatchSnapshot();
});
