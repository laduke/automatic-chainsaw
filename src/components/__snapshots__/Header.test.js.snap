exports[`test Description renders 1`] = `
<p>
  Hello, this is the description
</p>
`;

exports[`test FilterLinks render with feedback 1`] = `
<p>
  <span
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "teal",
      }
    }>
    show all 
  </span>
  <span
    onClick={[Function]}
    style={Object {}}>
    show favorites 
  </span>
</p>
`;

exports[`test FilterLinks render with feedback 2`] = `
<p>
  <span
    onClick={[Function]}
    style={Object {}}>
    show all 
  </span>
  <span
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "teal",
      }
    }>
    show favorites 
  </span>
</p>
`;

exports[`test Loading appears when buoy data is loading 1`] = `
<h4
  style={
    Object {
      "position": "fixed",
      "right": 10,
      "top": 0,
    }
  }>
   loading 
</h4>
`;

exports[`test Loading appears when buoy data is loading 2`] = `null`;

exports[`test Publication Date renders 1`] = `
<p>
  Fri, 09 Dec 2016 00:16:24 +0000
</p>
`;

exports[`test Title renders 1`] = `
<h3>
  NDBC - Station Observations near 40N 73W
</h3>
`;
